Lib/idlelib/run.py

```diff
@@ -271,7 +271,13 @@ def print_exc(typ, exc, tb):
                        "debugger_r.py", "bdb.py")
             cleanup_traceback(tbe, exclude)
             traceback.print_list(tbe, file=efile)
-        lines = get_message_lines(typ, exc, tb)
+        if not isinstance(exc, NameError) and not isinstance(exc, AttributeError):
+            lines = get_message_lines(typ, exc, tb)
+        else:
+            if "\n" in str(exc):
+                lines = [f"{typ.__name__}: {str(exc)}"]
+            else:
+                lines = get_message_lines(typ, exc, tb)
```

Comment:
After posting, I thought about doing the interception before calling get_message_lines.  If we were to use this idea, I much prefer.
```suggestion
        if ((not isinstance(exc, NameError) and not isinstance(exc, AttributeError))
            or "\n" in str(exc)):
            lines = get_message_lines(typ, exc, tb)
        else:  # User-created Name/AttributeError with multiline message, GH-135511. 
             lines = [f"{typ.__name__}: {str(exc)}"]
```
However, see the alternate change below.